### ISSUE TITLE
Got a crash when calling requestPermission with version 5.1.16

### ISSUE THREAD

**Issue by Joemiao120** [OneSignal/OneSignal-Android-SDK#2140]
### What happened?

I rebuild my project with Onesignal [5.0.0, 5.99.99], it crash when I open the app and init onesignl.

### Steps to reproduce?

```Markdown
1. Install v5.1.16
2. Add this piece of code when init
CoroutineScope(Dispatchers.IO).launch {
            OneSignal.Notifications.requestPermission(false)
        }
3. Open the app
```

### What did you expect to happen?

It works good without crashes.

### OneSignal Android SDK version

Release 5.1.16

### Android version

14

### Relevant log output

```Shell
FATAL EXCEPTION: DefaultDispatcher-worker-3 (Ask Gemini)
    Process: com.journeydigital.rymanresidentapp2, PID: 14066
    java.lang.reflect.InvocationTargetException
    	at java.lang.reflect.Constructor.newInstance0(Native Method)
    	at java.lang.reflect.Constructor.newInstance(Constructor.java:343)
    	at com.onesignal.common.services.ServiceRegistrationReflection.resolve(ServiceRegistration.kt:88)
    	at com.onesignal.common.services.ServiceProvider.getServiceOrNull(ServiceProvider.kt:79)
    	at com.onesignal.common.services.ServiceProvider.getService(ServiceProvider.kt:67)
    	at com.onesignal.common.services.ServiceRegistrationReflection.resolve(ServiceRegistration.kt:81)
    	at com.onesignal.common.services.ServiceProvider.getServiceOrNull(ServiceProvider.kt:79)
    	at com.onesignal.common.services.ServiceProvider.getService(ServiceProvider.kt:67)
    	at com.onesignal.common.services.ServiceRegistrationReflection.resolve(ServiceRegistration.kt:81)
    	at com.onesignal.common.services.ServiceProvider.getServiceOrNull(ServiceProvider.kt:79)
    	at com.onesignal.common.services.ServiceProvider.getService(ServiceProvider.kt:67)
    	at com.onesignal.internal.OneSignalImp.getNotifications(OneSignalImp.kt:493)
    	at com.onesignal.OneSignal.getNotifications(OneSignal.kt:62)
    	at com.journeydigital.rymanresidentapp.usecase.library.OneSignalWrapper$initOneSignal$1.invokeSuspend(OneSignalWrapper.kt:56)
    	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
    	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:104)
    	at kotlinx.coroutines.internal.LimitedDispatcher$Worker.run(LimitedDispatcher.kt:111)
    	at kotlinx.coroutines.scheduling.TaskImpl.run(Tasks.kt:99)
    	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:585)
    	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:802)
    	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:706)
    	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:693)
    	Suppressed: kotlinx.coroutines.internal.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@9821053, Dispatchers.IO]
    Caused by: java.lang.RuntimeException: Can't create handler inside thread Thread[DefaultDispatcher-worker-3,5,main] that has not called Looper.prepare()
    	at android.os.Handler.<init>(Handler.java:227)
    	at android.os.Handler.<init>(Handler.java:129)
    	at android.app.Activity.<init>(Activity.java:895)
    	at com.onesignal.core.internal.permissions.impl.RequestPermissionService.<init>(RequestPermissionService.kt:13)
    	... 22 more
```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

**Comment by lynaysereyvath** [comment]
I have the same issue from `OneSignal.initWithContext(this)` in application class.

**Comment by munibkhanali** [comment]
Hi 
I am also having this issue. downgrading to v 5.1.15 allows to continue the Development.

**Comment by Office-Murtaza** [comment]
> [5.0.0, 5.99.99],

please write the entire dependency for downgrading @munibkhanali .
i am currently using below 
 implementation 'com.onesignal:OneSignal:[5.0.0, 5.99.99]'

**Comment by munibkhanali** [comment]
Hi @Office-Murtaza 
Please try this

`implementation 'com.onesignal:OneSignal: 5.1.15'`

Thanks

**Comment by lynaysereyvath** [comment]
> Hi @Office-Murtaza Please try this
> 
> `implementation 'com.onesignal:OneSignal: 5.1.15'`
> 
> Thanks

This helps. Thanks

**Comment by najam-amfnrh9** [comment]
Hi @lynaysereyvath or @munibkhanali 
could you please share the complete sdk's implementations as i have still problem?
like what coroutines and one signals sdk your are using

**Comment by abudawud** [comment]
I have the same issue, downgrading to previous version resolve the problem
```
    implementation("com.onesignal:OneSignal:5.1.15")
```

**Comment by nan-li** [comment]
Hi everyone, thank you for reporting this. There was a change to speed up initialization of the SDK but is causing this crash. Please use v5.1.15 of the Android SDK while we issue a new release to remedy this.

**Comment by jinliu9508** [comment]
A fix is out with 5.1.17 to revert the breaking change that was merged in 5.1.16. Please update to the latest SDK version.
